### DISCUSSION TITLE
Is there an alternative to the view macro?

### DISCUSSION THREAD

**Discussion by flosse** [leptos-rs/leptos#366] [General]
There are advantages and disadvantages of macros over simple Rust (Builder Patterns).
I don't want to discuss the general topic here, but just ask if there are any plans to support an alternative to the macro "view" or if this is just a basic only option.

**Comment by gbj** [comment]
Yes; the current version supports a builder pattern, although it's not very well documented. Try expanding one of the view macros to see what it looks like.

**Reply by flosse** [reply]
Thanks!
I added an example (#373) so that others who have similar questions have an example right away.
